### ISSUE TITLE
Conflict over RPM lock when using SSM runners

### ISSUE THREAD

**Issue by ay0o** [philips-labs/terraform-aws-github-runner#3614]
When `enable_ssm_on_runners` is set to `true`, there is a probability that the installation of SSM conflicts with the installation of the other tools you install with `dnf`, resulting in a `rpm.lock` error:

```
user-data: RPM: error: can't create transaction lock on /var/lib/rpm/.rpm.lock (Resource temporarily unavailable)
```

This error causes cloud-init to end in a failed state, hence, not failing something along the process and not being able to schedule jobs, and eventually being terminated depending on the `minimum_running_time_in_minutes` value.

```
[FAILED] Failed to start: cloud-fina… Execute cloud user/final scripts.
```

This is specially noticeable when using ephemeral runners, because as a job failed to run on the scheduled runner, it will now keep waiting until another job launches a runner with the same labels, "stealing" this job's runner, and causing a snow ball effect down the line.

When ephemeral runners are not used, this is less of a concern, because there will be some idle runner waiting to pick up jobs.

**Comment by karmajunkie** [comment]
I'm also running into this problem. I can fix it on non-ephemeral runners by logging in and running the user-data script manually, but this is really problematic for obvious reasons.

Is the solution here to create our own ami's to use that have all the pieces in place at boot time?

**Comment by ay0o** [comment]
for now, we have decided to fix this from a higher, "logical" level.

A runner should really be a black box, you shouldn't need to enter at the OS level unless you are specifically debugging something, so we just removed the `enable_ssm_on_runners = true`.

If for any reason we would require to enter at the OS level to troubleshoot, we can enable it and disable again when we are done.

So far, no issues for the past week.
